js/init.js

```diff
@@ -0,0 +1,51 @@
+import dayjs from 'https://esm.archive.org/dayjs'
+import { isValidCustomElement, registerCustomElement } from './utils-wc.js'
+import {
+  getPropertiesFrom_DateConversion,
+  isContextRequest_DateConveresion,
+  isValidObject_DateConversion,
+} from './context-date.js'
+
+/**
+ * Our own components, but not loading them just yet.
+ */
+const OUR_COMPONENTS = [['bt-date', './bt-date.js']]
+
+const handleContextRequest_DateConversion = (event) => {
+  if (isContextRequest_DateConveresion(event)) {
+    event.stopPropagation()
+    const { date, dateHumanFormat } = getPropertiesFrom_DateConversion(event)
+    if (typeof date === 'string') {
+      const data = dayjs(date)
+      const dateUnix = data.unix()
+      const dateIsoString = data.toISOString()
+      const dateHuman = data.format(dateHumanFormat)
+      const payload = { date, dateIsoString, dateUnix, dateHuman }
+      isValidObject_DateConversion(payload) && event.callback(payload)
```

Comment:
Only callback IF validation passed. Hence &&


Probably we'd want also such a validator per component "contract".  Checking if has slot and attributes before customElements.define.  In the same way we want to make sure a context object is valid before passing it back